src/productcatalogservice/Dockerfile

```diff
@@ -12,29 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.15-alpine AS builder
-RUN apk add --no-cache ca-certificates git && \
-      wget -qO/go/bin/dep https://github.com/golang/dep/releases/download/v0.5.0/dep-linux-amd64 && \
-      chmod +x /go/bin/dep
+FROM golang:1.16-alpine AS builder
+RUN apk add --no-cache ca-certificates git
 
-ENV PROJECT github.com/GoogleCloudPlatform/microservices-demo/src/productcatalogservice
+ENV PROJECT github.com/GoogleCloudPlatform/cloud-ops-sandbox/src/productcatalogservice
 WORKDIR /go/src/$PROJECT
 
 # restore dependencies
-COPY Gopkg.* ./
-RUN dep ensure --vendor-only -v
-
-COPY . .
-RUN go build -o /productcatalogservice .
+COPY server.go go.mod go.sum ./
+COPY genproto ./genproto/
+RUN go mod vendor -v
+RUN go install .
 
 FROM alpine AS release
 RUN apk add --no-cache ca-certificates
 RUN GRPC_HEALTH_PROBE_VERSION=v0.3.5 && \
     wget -qO/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
     chmod +x /bin/grpc_health_probe
 WORKDIR /productcatalogservice
-COPY --from=builder /productcatalogservice ./server
+COPY --from=builder /go/bin/productcatalogservice /productcatalogservice
 COPY products.json .
 EXPOSE 3550
-ENTRYPOINT ["/productcatalogservice/server"]
+ENTRYPOINT ["/productcatalogservice"]
```

Comment:
When I try to deploy this using `make deploy`, I'm seeing this error: 
```
Error: failed to start container "server": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"/productcatalogservice\": permission denied": unknown
```

Isn't this a directory? Should it be `/productcatalogservice/productcatalogservice` or `./productcatalogservice` instead? That seemed to fix the issue for me

Reply:
@daniel-sanche ah, yes, you are right. I was too into the Go source code itself and didn't realize the issue here. Thanks for catching